src/umbraco.cms/businesslogic/Packager/PackageInstance/utill.cs

```diff
@@ -301,7 +301,37 @@ public static void ZipPackage(string Path, string savePath)
                     memoryStream.CopyTo(fileStream);
                 }
             }
+        }
+
+        /// <summary>
+        /// Zips the package.
+        /// </summary>
+        /// <returns>The package in a byte array.</returns>
+        /// <param name="Path">The path.</param>
+        public static byte[] ZipPackage(string Path)

```

Comment:
The other ZipPackage function is now unused, we should scrap it.
It would be more memory efficient to pass in the stream to save this to into this function as the destination. I don't know how big packages get but just sending the zipped data directly to the new file stream rather than into memory and then an array and back out would be better.